src/main/java/strategy/BalanceStrategy.java

```diff
@@ -0,0 +1,11 @@
+package strategy;
+
+import db.ShopStorage;
+import model.FruitTransaction;
+
+public class BalanceStrategy implements FruitShopStrategy {
+    @Override
+    public void doActivity(FruitTransaction transaction, ShopStorage storage) {
```

Comment:
this class is a handler, not a strategy, should implement OperationHandler as you had previously, with the only method `handle`